src/java.base/share/classes/sun/nio/ch/IOUtil.java

```diff
@@ -559,58 +559,22 @@ static long bufferAddress(ByteBuffer buf) {
         return NIO_ACCESS.getBufferAddress(buf);
     }
 
-    public static FileDescriptor newFD(int i) {
-        FileDescriptor fd = new FileDescriptor();
-        setfdVal(fd, i);
-        return fd;
-    }
-
-    static native boolean randomBytes(byte[] someBytes);
-
-    /**
-     * Returns two file descriptors for a pipe encoded in a long.
-     * The read end of the pipe is returned in the high 32 bits,
-     * while the write end is returned in the low 32 bits.
-     */
-    static native long makePipe(boolean blocking) throws IOException;
-
-    static native int write1(int fd, byte b) throws IOException;
-
-    /**
-     * Read and discard all bytes.
-     */
-    static native boolean drain(int fd) throws IOException;
-
-    /**
-     * Read and discard at most one byte
-     * @return the number of bytes read or IOS_INTERRUPTED
-     */
-    static native int drain1(int fd) throws IOException;
-
-    public static native void configureBlocking(FileDescriptor fd,
-                                                boolean blocking)
-        throws IOException;
-
     public static native int fdVal(FileDescriptor fd);
 
     static native void setfdVal(FileDescriptor fd, int value);
 
-    static native int fdLimit();
-
-    static native int iovMax();
+    private static native int iovMax();
 
-    static native long writevMax();
+    private static native long writevMax();
 
-    static native void initIDs();
+    private static native void initIDs();
 
     /**
      * Used to trigger loading of native libraries
      */
     public static void load() { }
```

Comment:
do we still need this method?

Reply:
Yes, it still needs to be called in a few places, e.g., for classes whose native component needs the `fdval()` and `handleval()` functions.

Reply:
that's a good point. The comment might need to be updated to reflect that.

Reply:
Comments added in 853d349.